centrallix/tests/test_cxss-crypto_05.c

```diff
@@ -0,0 +1,117 @@
+    assert(result == CXSS_CRYPTO_SUCCESS);
+    assert(ciphertext != NULL);
+    assert(cipherLen > 0);
+    assert(memcmp(data, ciphertext, strlen(data)) != 0);
+
+    /** test decryption for key 1 **/
+    char plaintext[4096];
+    int plainLen = 0;
+
+    result = cxssDecryptRSA(ciphertext, cipherLen, privKey1, privSize1, plaintext, &plainLen);
+    assert(result == CXSS_CRYPTO_SUCCESS);
+    assert(plaintext != NULL);
+    assert(plainLen > 0);
+    assert(strcmp(data, plaintext) == 0);
+
+    /** make sure different messages with same key are not the same **/
+    int cipherLen2 = 4096;
+    char ciphertext2[4096];
+    clearBuff(ciphertext, 4096);
+    clearBuff(ciphertext2, 4096);
+
+    result = cxssEncryptRSA(short1, strlen(short1), pubKey1, pubSize1, ciphertext, &cipherLen);
+    assert(result == CXSS_CRYPTO_SUCCESS);
+    result = cxssEncryptRSA(short2, strlen(short2), pubKey1, pubSize1, ciphertext2, &cipherLen2);
+    assert(result == CXSS_CRYPTO_SUCCESS);
+    assert(memcmp(ciphertext, ciphertext2, 4096) != 0);
+
+    /** make sure same data and same key do not match (padding) but decrypt correctly **/
+    clearBuff(ciphertext, 4096);
+    clearBuff(ciphertext2, 4096);
+    result = cxssEncryptRSA(short1, strlen(short1)+1, pubKey1, pubSize1, ciphertext, &cipherLen);
+    assert(result == CXSS_CRYPTO_SUCCESS);
+    result = cxssEncryptRSA(short1, strlen(short1)+1, pubKey1, pubSize1, ciphertext2, &cipherLen2);
+    assert(result == CXSS_CRYPTO_SUCCESS);
+    assert(memcmp(ciphertext, ciphertext2, 4096) != 0);
+
+    int plainLen2 = 4096;
+    char plaintext2[4096];
+
+    result = cxssDecryptRSA(ciphertext, cipherLen, privKey1, privSize1, plaintext, &plainLen);
+    assert(result == CXSS_CRYPTO_SUCCESS);
+    result = cxssDecryptRSA(ciphertext2, cipherLen2, privKey1, privSize1, plaintext2, &plainLen2);
+    assert(result == CXSS_CRYPTO_SUCCESS);
+    assert(strcmp(plaintext, plaintext2) == 0);
+
+
+    /** make sure same data and different key are not the same, but match on decrypt **/
+    clearBuff(ciphertext, 4096);
+    clearBuff(ciphertext2, 4096);
+    result = cxssEncryptRSA(short1, strlen(short1)+1, pubKey1, pubSize1, ciphertext, &cipherLen);
+    assert(result == CXSS_CRYPTO_SUCCESS);
+    result = cxssEncryptRSA(short1, strlen(short1)+1, pubKey2, pubSize2, ciphertext2, &cipherLen2);;
+    assert(result == CXSS_CRYPTO_SUCCESS);
+    assert(memcmp(ciphertext, ciphertext2, 4096) != 0);
+
+    result = cxssDecryptRSA(ciphertext, cipherLen, privKey1, privSize1, plaintext, &plainLen);
+    assert(result == CXSS_CRYPTO_SUCCESS);
+    result = cxssDecryptRSA(ciphertext2, cipherLen2, privKey2, privSize2, plaintext2, &plainLen2);
+    assert(result == CXSS_CRYPTO_SUCCESS);
+    assert(strcmp(plaintext, plaintext2) == 0);
+    assert(strcmp(plaintext, short1) == 0);
+
+
+    cxssCryptoCleanup();
+
+    return 0;
+    }
+
+void clearBuff(char* array, int length)
```

Comment:
You can use memset() or bzero() for this purpose.

Reply:
Got it! thanks for the tip